probe/pkg/runtime/runtime.go

```diff
@@ -0,0 +1,249 @@
+	}
+	defer response.Body.Close()
+	if response.StatusCode != http.StatusAccepted {
+		return errors.Errorf("Creation request to Central instance was not accepted: %s.", response.Status)
+	}
+
+	r.centralState = &central
+	err = r.ensureCentralState(constants.CentralRequestStatusReady.String())
+	return err
+}
+
+// Verify that the Central instance has the expected properties and that the
+// Central UI is reachable.
+func (r *Runtime) verifyCentral(ctx context.Context) error {
+	if r.centralState.InstanceType != fleetmanager.StandardInstanceType {
+		return errors.Errorf("Central has wrong instance type. Expected %s, got %s.", fleetmanager.StandardInstanceType, r.centralState.InstanceType)
+	}
+
+	err := r.pingURL(r.centralState.CentralUIURL)
+	return err
+}
+
+func (r *Runtime) pingURL(url string) error {
+	request, err := http.NewRequest("HEAD", url, nil)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to create request for Central UI %s", r.centralState.Id)
+	}
+	response, err := httpclient.HTTPClient.Do(request)
+	if err != nil {
+		return errors.Wrapf(err, "Central UI of instance %s not reachable", r.centralState.Id)
+	}
+	defer response.Body.Close()
+	if response.StatusCode != http.StatusOK {
+		return errors.Errorf("Central UI %s did not respond with status OK. Got: %s", r.centralState.Id, response.Status)
+	}
+	return nil
+}
+
+// Delete the Central instance and verify that it transitioned to 'deprovision' state.
+func (r *Runtime) deleteCentral(ctx context.Context) error {
+	response, err := r.fleetManagerClient.DeleteCentralById(ctx, r.centralState.Id, true)
+	glog.Infof("Deletion of Central instance %s requested.", r.centralState.Id)
+	if err != nil {
+		return errors.Wrapf(err, "Deletion of Central instance %s failed", r.centralState.Id)
+	}
+	defer response.Body.Close()
+	if response.StatusCode != http.StatusAccepted {
+		return errors.Errorf("Deletion request to Central instance %s was not accepted: %s.", r.centralState.Id, response.Status)
+	}
+
+	err = r.ensureCentralState(constants.CentralRequestStatusDeprovision.String())
+	return err
+}
+
+func newCentralName() (string, error) {
+	rnd := make([]byte, 8)
+	_, err := rand.Read(rnd)
+	if err != nil {
+		return "", errors.Wrapf(err, "Reading random bytes for unique central name")
+	}
+	rndString := hex.EncodeToString(rnd)
+
+	return fmt.Sprintf("probe-%s", rndString), nil
+}
+
+func (r *Runtime) ensureCentralState(targetState string) error {
+	ctxTimeout, cancel := context.WithTimeout(context.Background(), r.config.RuntimePollTimeout)
+	defer cancel()
+
+	isDone := make(chan bool, 1)
+
+	go r.pollCentral(ctxTimeout, isDone, targetState)
+
+	select {
+	case <-ctxTimeout.Done():
+		return errors.Wrapf(ctxTimeout.Err(), "Central instance %s did not reach %s state", r.centralState.Id, targetState)
+	case <-isDone:
+		return nil
+	}
+}
+
+func (r *Runtime) pollCentral(ctx context.Context, isDone chan bool, targetState string) error {
```

Comment:
Just a stylistic remark that will likely be obsolete after addressing comments: when passing channels between functions, it's common to be as specific as possible about the operations that will happen on them. Since this method only writes to the channel but will never read from it, one would write this as
```suggestion
func (r *Runtime) pollCentral(ctx context.Context, isDone chan<- bool, targetState string) error {
```